src/bouncerbot/index.ts

```diff
@@ -1,55 +1,37 @@
-import { getMessageStream, sendMessage } from '../utils'
+import { getMessageStream, sendMessage, getRandomItemFromList } from '../utils'
 import { ListenResponse } from '@hoprnet/hopr-protos/node/listen_pb'
 import { Message } from '../message'
+import response from './response.json'
 
 
-const directory = {}
-
-const messages = [
-  'Oh, you here for the DAI party?',
-  'I see. I don’t see you in the guest list. Only people on the guest list get past me. Get out of here.',
-  'I don’t know what to tell you. If you ain’t on the list, you ain’t coming in',
-  'This party’s for social media influencers only. Scram',
-  'I don’t know what to tell you. I’m sure if you hang around long enough, someone will help you out'
-] 
+enum NodeStates {
+  IsHinted,
+  RequiresProof,
+  InGuestList,
+}
 
-export default async (hoprAddress) => {
-  const botName = '🥊 Bouncerbot (v2)'
-  console.log(`${botName} has been added`);
+class Bouncebot {
```

Comment:
@nionis I think the work @shresthagrawal is doing here is great, particularly abstracting the bot mechanic in an class. In #10 I'm describing some generic methods we could implement, but I'm open to any interface. For me, #12 might be trickier tho, as we want to do something like `pay(address: HOPRAddress)` which _somehow_ triggers the SDK. It would definitely require some Ethereum gateway access, do we have something already implemented in `hopr-ethereum` we could reuse here?

Reply:
@jjperezaguinaga is the goal here to pay in `HOPR` or `DAI`?

Reply:
@nionis DAI for the time being.

Reply:
@jjperezaguinaga `hopr-ethereum` cannot help with that as it only provides the ABIs and addresses for our own contracts, it also generates `IERC20` abi but it's not part of our package.